libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/prompts/UserTokenAccess.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MT License.
+
+package com.microsoft.bot.dialogs.prompts;
+
+import com.microsoft.bot.builder.ConnectorClientBuilder;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.connector.ConnectorClient;
+import com.microsoft.bot.connector.authentication.ClaimsIdentity;
+import com.microsoft.bot.connector.authentication.ConnectorFactory;
+import com.microsoft.bot.connector.authentication.UserTokenClient;
+import com.microsoft.bot.schema.SignInResource;
+import com.microsoft.bot.schema.TokenExchangeRequest;
+import com.microsoft.bot.schema.TokenResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class UserTokenAccess {
+
+    private UserTokenAccess() {
+    }
+
+    public static CompletableFuture<TokenResponse> getUserToken(
+        TurnContext turnContext,
+        OAuthPromptSettings settings,
+        String magicCode) {
+
+        UserTokenClient userTokenClient = turnContext.getTurnState().get(UserTokenClient.class);
+
+        if (userTokenClient != null) {
+            return userTokenClient.getUserToken(
+                turnContext.getActivity().getFrom().getId(),
+                settings.getConnectionName(),
+                turnContext.getActivity().getChannelId(),
+                magicCode);
+        } else {
```

Comment:
Out of curiosity, is there any reason why the other condition was not ported? (Of course, with `UserTokenProvider` instead as `IExtendedUserTokenProvider` does not exist on Java)
```
else if (turnContext.Adapter is IExtendedUserTokenProvider adapter)
 {
    return await adapter.GetUserTokenAsync(turnContext, settings.OAuthAppCredentials, settings.ConnectionName, magicCode, cancellationToken).ConfigureAwait(false);
}
```

Also, I found [this](https://github.com/southworks/botbuilder-java/blob/5bcceebc0b824b419a9fd8068233faef6ef015ce/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/teams/TeamsSSOTokenExchangeMiddleware.java#L165-L176) and [this](https://github.com/southworks/botbuilder-java/blob/5bcceebc0b824b419a9fd8068233faef6ef015ce/libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/SkillDialog.java#L374-L382) snippets that have a few discrepancies [with C#](https://github.com/microsoft/botbuilder-dotnet/blob/3a750837ef3e1da44861e0b7ec562a94583d1051/libraries/Microsoft.Bot.Builder/Teams/TeamsSSOTokenExchangeMiddleware.cs#L136-L158).